daemon/containerd/image_manifest.go

```diff
@@ -78,21 +124,40 @@ func (im *ImageManifest) Metadata() containerdimages.Image {
 	return md
 }
 
-// IsPseudoImage returns false if the manifest has no layers or any of its layers is a known image layer.
-// Some manifests use the image media type for compatibility, even if they are not a real image.
-func (im *ImageManifest) IsPseudoImage(ctx context.Context) (bool, error) {
-	desc := im.Target()
-
+func (im *ImageManifest) IsAttestation() bool {
 	// Quick check for buildkit attestation manifests
 	// https://github.com/moby/buildkit/blob/v0.11.4/docs/attestations/attestation-storage.md
 	// This would have also been caught by the layer check below, but it requires
 	// an additional content read and deserialization of Manifest.
-	if _, has := desc.Annotations[attestation.DockerAnnotationReferenceType]; has {
+	if _, has := im.Target().Annotations[attestation.DockerAnnotationReferenceType]; has {
+		return true
+	}
+	return false
+}
+
+// IsPseudoImage returns false when any of the below is true:
+// - The manifest has no layers
+// - None of its layers is a known image layer.
+// - The manifest has unknown/unknown platform.
+//
+// Some manifests use the image media type for compatibility, even if they are not a real image.
+func (im *ImageManifest) IsPseudoImage(ctx context.Context) (bool, error) {
+	if im.IsAttestation() {
```

Comment:
Ever so slightly on the fence if we should explicitly check for `IsAttestation` in locations where we call `IsPseudoImage()` - that's fine for a follow-up, but I wondered if it all became too much "magic" (also related to the comment further up).

Thinking of locations like https://github.com/moby/moby/blob/59996a493cfc36871ddd8f6ede525def98b69e2f/daemon/containerd/image_exporter.go#L284-L285


If we should make that 

```go
if IsAttestation {
    return true
}
if IsPseudoImage {
....
```

Reply:
I think it's fine for a follow-up; I was considering that _possibly_ it would be more transparent on the call-site to indicate what we're skipping (attestations .. or other "non-image" content).

Reply:
In that `image_exporter.go` case we're filtering out manifests before unpacking - we don't want to just explicitly skip attestations, we want to only unpack runnable images that have supported layer types. So in this case it doesn't make sense to make an explicit check for attestation IMO.